src/auth/service.py

```diff
@@ -0,0 +1,143 @@
+"""Authentication service."""
+
+from fastapi import HTTPException, status
+from keycloak.exceptions import KeycloakAuthenticationError
+
+from auth.config import keycloak_openid
+from auth.logger import L
+from auth.models import ProjectContext, TokenInfo, TokenPair, TokenPairInput, UserInfo
+from auth.util import get_timestamp_now
+
+
+def authenticate_user(username: str, password: str) -> str:
+    """Authenticate the user using Keycloak and return an access token.
+
+    Args:
+        username: The username of the user.
+        password: The password of the user.
+
+    Returns:
+        The access token of the user.
+    """
+    try:
+        token = keycloak_openid.token(username, password)
```

Comment:
I wonder, what's the use case for this: testing by using Keycloak local accounts or something else?

Reply:
I added it in the beginning but it is not used anywhere. And yes, it was for allowing to authenticate using local accounts, but it shouldn't be part of this PR.